### PR TITLE
Introduce new configurations package

### PR DESCRIPTION
**Pull request by jastang**
### Description of your changes
This PR supports two new GET service endpoints for Upbound root configurations.

`GET /v1/configurations/{accountName}`
`GET /v1/configurations/{accountName}/{cfgName}`

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
- adapted some of the examples in this repository for `Get` and `List` Configurations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,11 @@ development and subject to breaking changes.
 
 The following services are currently supported:
 - Accounts
+- Configurations
 - Control Planes
+- Organizations
+- Repositories
+- Robots
```
Comment (hasheddan) [inline README.md:14]: Thanks for the updates here!

File: service/configurations/client.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	*up.Config
+}
+
+// NewClient builds a configurations client from the given config.
+func NewClient(cfg *up.Config) *Client {
+	return &Client{cfg}
+}
+
+// List all configurations for an account on Upbound.
+func (c *Client) List(ctx context.Context, account string) (*ConfigurationListResponse, error) {
```
Comment (hasheddan) [inline service/configurations/client.go:40]: @jastang do you anticipate adding create / update functionality in a separate PR?

File: service/configurations/types.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+package configurations
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// ConfigurationResponse represents the concept of a Configuration Upbound.
```
Comment (hasheddan) [inline service/configurations/types.go:23]:
```suggestion
// ConfigurationResponse represents the concept of a Configuration on Upbound.
```

File: service/configurations/client.go (hunk 2 above)
Comment (jastang) [reply service/configurations/client.go:40]: @hasheddan yep! I think they're also stable but might still have some in-flight changes in the near future, so I opted to introduce them separately.

**Comment by jastang** [comment]
Looks like we hit a brownout period on GitHub today. Will update the workflows shortly

**Review by hasheddan — APPROVED** [review]

**Pull request merged**